pkg/distributor/distributor.go

```diff
@@ -91,6 +92,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	flag.DurationVar(&cfg.ClientCleanupPeriod, "distributor.client-cleanup-period", 15*time.Second, "How frequently to clean up clients for ingesters that have gone away.")
 	flag.Float64Var(&cfg.IngestionRateLimit, "distributor.ingestion-rate-limit", 25000, "Per-user ingestion rate limit in samples per second.")
 	flag.IntVar(&cfg.IngestionBurstSize, "distributor.ingestion-burst-size", 50000, "Per-user allowed ingestion burst size (in number of samples).")
+	flag.BoolVar(&cfg.CompressPushes, "distributor.compress-pushes", false, "Compress sample data pushed to ingesters.")
```

Comment:
The var name and description are too specific. If the flag is set, the compression is applied to all ingester traffic, not just sample pushes.